### PR TITLE
Sync to latest XRT master branch

### PR DESCRIPTION
**Pull request by maxzhen**
This PR is to implement new command chaining interface implemented by https://github.com/Xilinx/XRT/pull/8194.

### CODE REVIEW

File: src/driver/amdxdna/aie2_message.c [hunk 1]
```diff
@@ -520,9 +521,10 @@ int aie2_execbuf(struct amdxdna_hwctx *hwctx, struct amdxdna_sched_job *job,
 		XDNA_DBG(xdna, "Invalid ERT cmd op code: %d", op);
 		return -EINVAL;
 	}
-	msg.handle = handle;
+	msg.handle = job;
 	msg.notify_cb = notify_cb;
 	msg.send_data = (u8 *)&req;
+	/* TODO: no need to fill msg.send_size? */
```
Comment (mamin506) [inline src/driver/amdxdna/aie2_message.c]: msg.send_size and msg.opcode are depending on 'op'. Thus it was filled in the switch block.

File: src/driver/amdxdna/aie2_message.c [hunk 2]
```diff
@@ -534,141 +536,216 @@ int aie2_execbuf(struct amdxdna_hwctx *hwctx, struct amdxdna_sched_job *job,
...
+		offset += size;
+	}
+
+	aie2_cmdlist_prepare_request(&req, cmdbuf_abo, size, payload->command_count);
+
+	msg.handle = job;
 	msg.notify_cb = notify_cb;
 	msg.send_data = (u8 *)&req;
 	msg.send_size = sizeof(req);
+	ret = xdna_mailbox_send_msg(chann, &msg, TX_TIMEOUT);
```
Comment (mamin506) [inline src/driver/amdxdna/aie2_message.c:722]: msg.opcode is not set??
Maybe
```suggestion
        if (op == ERT_START_CU)
             msg.opcode = MSG_OP_CHAIN_EXEC_BUFFER_CF;
        else
            msg.opcode = MSG_OP_CHAIN_EXEC_DPU;
	ret = xdna_mailbox_send_msg(chann, &msg, TX_TIMEOUT);
```

**Review by mamin506 — COMMENTED** [review]
Overall this looks good to me. I will build and try this PR on tomorrow.

File: src/driver/amdxdna/aie2_message.c (hunk 1 above)
Comment (maxzhen) [reply src/driver/amdxdna/aie2_message.c]: OK.

File: src/driver/amdxdna/aie2_message.c (hunk 2 above)
Comment (maxzhen) [reply src/driver/amdxdna/aie2_message.c:722]: Fixed.

**Review by mamin506 — APPROVED** [review]

**Pull request merged**